Lesson_01/05Lesson/task01.cs

```diff
@@ -0,0 +1,127 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Lessons._05Lesson
+{
+    /// <summary>
+    /// Создать класс рациональных чисел. В классе два поля – числитель и знаменатель. 
+    /// Предусмотреть конструктор. Определить операторы ==, != (метод Equals()), <, >, 
+    /// <=, >=, +, –, ++, --. Переопределить метод ToString() для вывода дроби. 
+    /// Определить операторы преобразования типов между типом дробь,float, int. 
+    /// Определить операторы *, /, %.
+    /// </summary>
+    public class Digits : IComparable<Digits>
+    {
+        public int numerator { get; set; }
+        public int denominator { get; set; }
+
+
+        public Digits() { }
+        public Digits(int num, int den)
+        {
+            numerator = num;
+            denominator = den;
+        }
+
+        // override methods
+        public override int GetHashCode() => this.ToString().GetHashCode();
+        public override bool Equals(object? obj) => obj.ToString() == this.ToString();
+
+        public int CompareTo(Digits? other)
+        {
+            if ((this.numerator * other.denominator) > (other.numerator * this.denominator) && (this.denominator * other.denominator) == (other.denominator * this.denominator)) return 1;
+            if ((this.numerator * other.denominator) < (other.numerator * this.denominator) && (this.denominator * other.denominator) == (other.denominator * this.denominator)) return -1;
+            else return 0;
+        }
+
+        // переопределем операторы
+        public static bool operator ==(Digits d1, Digits d2) => d1.Equals(d2);
+        public static bool operator !=(Digits d1, Digits d2) => !d1.Equals(d2);
```

Comment:
```suggestion
        public static bool operator ==(Digits d1, Digits d2) => Equals(d1, d2);
        public static bool operator !=(Digits d1, Digits d2) => !(d1 != d2);
```

Если в d1 был бы null, то была бы проблема